src/hermes/commands/deposit/invenio_rdm.py

```diff
@@ -0,0 +1,717 @@
+
+    An API endpoint (usually ``/api/vocabularies/licenses``) can be used to check whether a given
+    license is supported by the Invenio instance. This function tries to retrieve the
+    license by the identifier at the end of the license URL path. If this identifier
+    does not exist on the Invenio instance, a :class:`RuntimeError` is raised. If no
+    license is given in the CodeMeta, ``None`` is returned.
+    """
+
+    license_url = ctx["codemeta"].get("license")
+
+    if license_url is None:
+        return None
+
+    if not isinstance(license_url, str):
+        raise RuntimeError(
+            "The given license in CodeMeta must be of type str. "
+            "Licenses of type 'CreativeWork' are not supported."
+        )
+
+    # First try: Look up license by assuming lower-case name is the correct identifier
+    parsed_url = urlparse(license_url)
+    url_path = parsed_url.path.rstrip("/")
+    license_id = url_path.split("/")[-1].lower()
+
+    response = requests.get(
+        f"{license_api_url}/{license_id}", headers={"User-Agent": hermes_user_agent}
+    )
+    if response.ok:
+        license_info = response.json()
+
+    # Second try: Fetch full list of licenses available... maybe we should cache this.
+    else:
+        license_info = _look_up_license_info(license_api_url, license_url)
+
+    return license_info["id"]
+
+
+def _look_up_license_info(license_api_url, license_url):
+    """Deliberately try to resolve the license URL to a valid InvenioRDM license information record from the
+    vocabulary.
+
+    First, this method tries to find the license URL in the list of known license vocabulary (which is fetched each
+    time, ouch...).
+
+    If the URL is not found (what is pretty probable by now, as CFFConvert produces SPDX-URLs while InvenioRDM still
+    relies on the overhauled opensource.org URLs), the SPDX information record is fetched and all valid cross references
+    are sought for.
+
+    :param license_api_url: Base API endpoint for InvenioRDM license vocabulary queries.
+    :param license_url: The URL for the license we are search an identifier for.
+    :return: The vocabulary record that is provided by InvenioRDM.
+    """
+    response = requests.get(
+        f"{license_api_url}?size=1000", headers={"User-Agent": hermes_user_agent}
+    )
+    response.raise_for_status()
+    valid_licenses = response.json()
+
+    def _search_license_info(_url):
+        for license_info in valid_licenses['hits']['hits']:
+            try:
+                if license_info['props']['url'] == _url:
+                    return license_info
+            except KeyError:
+                continue
+        else:
+            return None
+
+    license_info = _search_license_info(license_url)
+    if license_info is None and license_url.startswith('https://spdx.org/licenses/'):
+        response = requests.get(f"{license_url}.json", headers={"User-Agent": hermes_user_agent})
```

Comment:
Content negotiation would be nice. But they don't seem to support it 🙄 😅

Reply:
Yes, unfortunate they use the different URLs. However, it's at least documented... even though I could not find the source 😅